internal/agent/device/config/controller.go

```diff
@@ -33,10 +35,26 @@ func NewController(
 	}
 }
 
+func (c *Controller) init(ctx context.Context, current *v1alpha1.RenderedDeviceSpec) {
+	if current == nil {
+		return
+	}
+	currentIgnition, err := parseAndConvertConfig(lo.FromPtr(current.Config))
+	if err != nil {
+		c.log.Warnf("failed to parse current ignition: %+v", err)
+		return
+	}
+	lo.ForEach(currentIgnition.Storage.Files, func(f ignv3types.File, _ int) { c.hookManager.OnAfterReboot(ctx, f.Path) })
```

Comment:
I am not a fan of all of this `lo` package usage I understand it is helpful but it hides the actual logic.  Which to me defeats its purpose. Please use go code and if we need helper methods add to pkg/. The agent should not depend on 3rd party packages to perform simple operations.

Reply:
OK. Will change.